sda-commons-server-dropwizard/src/main/java/org/sdase/commons/server/dropwizard/bundles/GenericLookupYamlConfigurationFactory.java

```diff
@@ -0,0 +1,105 @@
+import io.dropwizard.configuration.YamlConfigurationFactory;
+import java.io.IOException;
+import java.util.List;
+import javax.annotation.Nullable;
+import javax.validation.Validator;
+import org.apache.commons.text.lookup.StringLookup;
+import org.sdase.commons.server.dropwizard.bundles.JacksonTypeScanner.MappableField;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A {@link YamlConfigurationFactory} that consumes all environment variables that begin with a
+ * configurable {@code propertyPrefix} (Dropwizard defaults to {@code dw.}) and uses them as
+ * overrides.
+ *
+ * @param <T> the type of the configuration objects to produce
+ */
+public class GenericLookupYamlConfigurationFactory<T> extends YamlConfigurationFactory<T> {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(GenericLookupYamlConfigurationFactory.class);
+
+  private final Class<T> configurationClass;
+  private final JacksonTypeScanner jacksonTypeScanner;
+  private final StringLookup lookup;
+
+  /**
+   * Creates a new configuration factory for the given class.
+   *
+   * @param configurationClass the configuration class
+   * @param validator the validator to use
+   * @param objectMapper the Jackson {@link ObjectMapper} to use
+   * @param propertyPrefix the system property name prefix used by overrides
+   * @param jacksonTypeScanner the scanner to derive available properties
+   * @param lookup the {@link StringLookup} to lookup values for {@link
+   *     MappableField#getEnvironmentVariableName()}
+   */
+  public GenericLookupYamlConfigurationFactory(
+      Class<T> configurationClass,
+      @Nullable Validator validator,
+      ObjectMapper objectMapper,
+      String propertyPrefix,
+      JacksonTypeScanner jacksonTypeScanner,
+      StringLookup lookup) {
+    super(configurationClass, validator, objectMapper, propertyPrefix);
+    this.configurationClass = configurationClass;
+    this.jacksonTypeScanner = jacksonTypeScanner;
+    this.lookup = lookup;
+  }
+
+  @Override
+  protected T build(JsonNode node, String path) throws IOException, ConfigurationException {
+    List<MappableField> mappableFields = jacksonTypeScanner.scan(configurationClass);
+    mappableFields.stream()
+        .filter(this::canLookup)
+        .forEach(mappableField -> configure(node, mappableField));
+    return super.build(node, path);
+  }
+
+  private boolean canLookup(MappableField mappableField) {
+    boolean canLookup = lookup.lookup(mappableField.getEnvironmentVariableName()) != null;
+    if (LOG.isInfoEnabled()) {
+      if (canLookup) {
+        LOG.info(
+            "{} of type {} picked from lookup.",
+            mappableField.getEnvironmentVariableName(),
+            mappableField.getPropertyTypeSimpleName());
+      } else {
+        LOG.info(
+            "{} of type {} not set.",
+            mappableField.getEnvironmentVariableName(),
+            mappableField.getPropertyTypeSimpleName());
+      }
+    }
```

Comment:
Maybe remove logging before merging, users can use the `config` command to get information.